nncf/quantization/algorithms/weight_compression/weight_lowering.py

```diff
@@ -260,21 +263,52 @@ def calculate_quantized_weight(
     :param config: Weight compression configuration.
     :return: Quantized weight tensor of uint8 type.
     """
-    if weight.dtype != TensorDataType.float32:
-        weight = weight.astype(TensorDataType.float32)
-    if scale.dtype != TensorDataType.float32:
-        scale = scale.astype(TensorDataType.float32)
 
-    num_bits = config.num_bits
+    if weight.backend == TensorBackend.numpy and not is_openvino_available():
+        log_once(logging.INFO, "Compression time may improve after installing OpenVINO")
 
-    level_low = 0
-    level_high = 2**num_bits - 1
+    if weight.backend == TensorBackend.numpy and is_openvino_available():
+        from nncf.openvino.quantization.compression_primitives import OV_COMPRESSION_PRIMITIVE_CACHE
+
+        compress_weight_primitive = OV_COMPRESSION_PRIMITIVE_CACHE.get_compress_weight_primitive(
+            config, weight.shape, scale.shape, zero_point.shape
+        )
+        compressed_weights = Tensor(compress_weight_primitive(weight.data, scale.data, zero_point.data))
```

Comment:
I would suggest to hide caching logic and have simple function call like `compress_weight` or `quantize_weight`.
```suggestion
        from nncf.openvino.quantization.weight_lowering import compress_weight
        
        compressed_weights = compress_weight(weight, scale, zero_point)
```

Reply:
Do you suggest to create a `quantize_weight()`/`compress_weight()` function alongside `calculate_quantized_weight()` function at `weight_lowering.py` and call the former one from the latter one? If so, don't you think it is an unnecessary function multiplication? In my opinion `weight_lowering.py` already has so many different function with similar names, it's hard to distinguish which does what.

Reply:
I tend to agree with your opinion. I'm working on it right now. If you have any proposal to refactor `weight_lowering.py`, please, let me know. Regarding this comment I hope you can extend functionality of `weight_lowering.py` without complication.